recipes/libproperties/all/conanfile.py

```diff
@@ -0,0 +1,47 @@
+from conans import ConanFile, CMake, tools
+
+
+class LibpropertiesConan(ConanFile):
+    name = "libproperties"
+    version = "1.1.2"
+    license = "Apache-2.0 License"
+    author = "libbylg@126.com"
+    url = "https://github.com/tinyhubs/libproperties"
+    description = "libproperties is a library to parse the Java .properties files. It was writen in pure C. And fullly compatible with the Java .properties file format."
+    topics = (".properties", "Java", "pure C")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False]}
+    default_options = {"shared": False}
+    generators = "cmake"
+
+    def source(self):
+        self.run("git clone https://github.com/tinyhubs/libproperties.git")
+        # This small hack might be useful to guarantee proper /MT /MD linkage
+        # in MSVC if the packaged project doesn't have variables to set it
+        # properly
+        tools.replace_in_file("libproperties/CMakeLists.txt", "PROJECT(libproperties)",
+                              '''PROJECT(libproperties)
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
```

Comment:
Please use a release, or if one is not available a particular git commit (and download it as a tarball/zip).

Also, to avoid patching, you can include the project as a subproject (using add_subdirectory).
See other recipes in this repo how to do this.

Reply:
Looking at it, tinyhubs is a fork of https://github.com/libbylg/libproperties .

Maybe use that one?

Reply:
And that one has a release!
https://github.com/libbylg/libproperties/releases

Reply:
Thanks for your help.
And `libbylg/libproperties` is the origin one, I'm ready remove it later.
I will add a release for `tinyhubs/libproperties` next weeks.

Reply:
I have make a release, and how to download from the recipe script?

Reply:
Please look at the other recipes:
- `config.yml` for listing all versions
- `all/conandata.yml` with url+sha256 (+eventual patches)
- `all/CMakeLists.txt` with a little cmake wrapper script.
- `all/conanfile.py` with the actual script.
It should:
   * do
        ```python
            def source(self):
                tools.get(**self.conan_data["sources"][self.version])
                os.rename("Python-{}".format(self.version), self._source_subfolder)
        ```
        to fetch the sources.
    *  contain`generators="cmake"` 
    * have a `_conifigure_cmake` to cache the cmake object.

But please look at other cmake recipes.
e.g. bullet3

Reply:
Where to retrive the sha256 of the release package? Or need to generate manual?

Reply:
Run `sha256sum` on the archive.
Or enter some nonsense, run conan and copy the hash that conan computes for you.

Reply:
An other question, should we used the zip file or the .tar.gz one or any of them?

Reply:
Any of them.
A lot recipes prefer the tar ones.

Reply:
Actually, I am warried about that  .tar.gz  package can not be uncompressed on windows or not.

Reply:
Python has tar + gunzip support builtin.
So no need to worry.
Unless you have very weird files in your releases.

There are a lot of recipes using the tarballs on windows without issue.